hbase-server/src/main/java/org/apache/hadoop/hbase/tool/Canary.java

```diff
@@ -119,8 +119,35 @@
  * </li>
  * </ol>
  */
-@InterfaceAudience.Private
-public final class Canary implements Tool {
+@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.TOOLS)
+public class Canary implements Tool, CanaryInterface {
```

Comment:
Can Canary be an Interface?

Reply:
Yeah please make the interface Canary and the tool CanaryTool or CanaryImpl or something.

Reply:
@saintstack  @busbey  Addressed your comments in latest commit. Could you please review. Thank you !